src/ScriptCs.Hosting/ScriptServicesBuilder.cs

```diff
@@ -131,6 +133,18 @@ public IScriptServicesBuilder Debug(bool debug = true)
             return this;
         }
 
+        public IScriptServicesBuilder SetOverride<TContract, TImpl>(TImpl value) where TImpl : TContract
+        {
+            Overrides[typeof(TContract)] = value;
+            return this;
+        }
+
+        public IScriptServicesBuilder SetOverride<TContract, TImpl>() where TImpl : TContract
+        {
+            Overrides[typeof(TContract)] = typeof(TImpl);
```

Comment:
You can set for the value either an instance or an implementation type.
Both are supported.
On Mon, Apr 6, 2015 at 11:05 PM Adam Ralph <notifications@github.com> wrote:

> In src/ScriptCs.Hosting/ScriptServicesBuilder.cs
> <https://github.com/scriptcs/scriptcs/pull/1016#discussion_r27853247>:
>
> > @@ -131,6 +133,18 @@ public IScriptServicesBuilder Debug(bool debug = true)
> >              return this;
> >          }
> >
> > +        public IScriptServicesBuilder SetOverride<TContract, TImpl>(TImpl value) where TImpl : TContract
> > +        {
> > +            Overrides[typeof(TContract)] = value;
> > +            return this;
> > +        }
> > +
> > +        public IScriptServicesBuilder SetOverride<TContract, TImpl>() where TImpl : TContract
> > +        {
> > +            Overrides[typeof(TContract)] = typeof(TImpl);
>
> @glennblock <https://github.com/glennblock> how does this work? Is it
> possible to set not only a value for a given contract type (as in the above
> overload) but also just an implementation type?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scriptcs/scriptcs/pull/1016/files#r27853247>.
>